plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MappedNamespaceContext.java

```diff
@@ -29,28 +28,30 @@
 
   private final Map<String, String> prefixMapping = new HashMap<>();
 
-  public MappedNamespaceContext(String prefix, String namespaceUri) {
-    checkMapping(prefix, namespaceUri);
+  public MappedNamespaceContext() {
     prefixMapping.put("xml", XMLConstants.XML_NS_URI);
-    prefixMapping.put(prefix, namespaceUri);
   }
 
-  public MappedNamespaceContext(Map<String, String> prefixMapping) {
-    for (Entry<String, String> entry : prefixMapping.entrySet()) {
-      checkMapping(entry.getKey(), entry.getValue());
-    }
-    this.prefixMapping.put("xml", XMLConstants.XML_NS_URI);
-    this.prefixMapping.putAll(prefixMapping);
+  public MappedNamespaceContext(String prefix, String namespaceUri) {
+    this();
+    addMapping(prefix, namespaceUri);
```

Comment:
It's a bit dangerous to call a public-non-final method from a constructor. Perhaps share a private method between them.